butane_core/tests/migration.rs

```diff
@@ -137,3 +137,275 @@ fn stable_add_column_alpha_order() {
         ]
     );
 }
+
+#[test]
+fn add_table_fkey() {
+    let known_int_type = DeferredSqlType::KnownId(TypeIdentifier::Ty(SqlType::Int));
+
+    let old = ADB::default();
+    let mut new = ADB::default();
+    let mut table_a = ATable::new("a".to_owned());
+    let column = AColumn::new(
+        "id".to_owned(),
+        known_int_type.clone(),
+        false, // nullable
+        true,  // pk
+        false, // auto
+        false, // unique
+        None,  // default
+    );
+    table_a.add_column(column);
+    new.replace_table(table_a.clone());
+
+    let mut table_b = ATable::new("b".to_owned());
+    let column = AColumn::new(
+        "b".to_owned(),
+        DeferredSqlType::Deferred(TypeKey::PK("a".to_owned())),
+        false, // nullable
+        true,  // pk
+        false, // auto
+        false, // unique
+        None,  // default
+    );
+    table_b.add_column(column);
+    new.replace_table(table_b.clone());
+
+    new.resolve_types().unwrap();
+
+    let mut resolved_table_b = ATable::new("b".to_owned());
+    let column = AColumn::new(
+        "b".to_owned(),
+        known_int_type.clone(),
+        false, // nullable
+        true,  // pk
+        false, // auto
+        false, // unique
+        None,  // default
+    );
+    resolved_table_b.add_column(column);
+
+    let ops = diff(&old, &new);
+
+    assert_eq!(
+        ops,
+        vec![
+            Operation::AddTable(table_a),
+            Operation::AddTable(resolved_table_b.clone()),
+        ]
+    );
+}
+
+fn create_add_renamed_table_fkey_ops() -> (Vec<Operation>, ADB, ATable, ATable) {
```

Comment:
Could you add a method comment here just to summarize what this helper is doing?